### PR TITLE
fix(datasets): Relax pandas.HDFDataSet dependencies which are broken on Windows

### LINKED ISSUE #402: kedro-datasets[pandas] install fails on windows
## Description

``pip install kedro-datasets[pandas]`` badly fails on Windows with an ugly stracktrace. 

## Context

I've been upgrading my old projects to finally use kedro-datasets. The recommended way to install ``pandas`` dependency (e.g. to read csv files with ``CSVDataset``) is to use ``pip install kedro-datasets[pandas]`` instead of ``pip install kedro-datasets pandas``. However the first command crashes badly while the second work, which goes against what is advised in the documentation. 

## Steps to Reproduce

Create a fresh conda env: 

```bash
conda create -n bug-kedro-pandas python=3.10 -y
conda activate bug-kedro-pandas
pip install kedro==0.18.14
pip install kedro-datasets
```
Now this fails (see stacktrace below): 
```bash
pip install kedro-datasets[pandas]
```

while this works: 

```bash
pip install pandas
```
## Expected Result

Install should work.

## Actual Result

```bash
(bug-kedro-pandas) C:\Users\>pip install kedro-datasets[pandas]
Requirement already satisfied: kedro-datasets[pandas] in c:\users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages (1.7.1)
Requirement already satisfied: kedro>=0.16 in c:\users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages (from kedro-datasets[pandas]) (0.18.14)
Requirement already satisfied: lazy-loader in c:\users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages (from kedro-datasets[pandas]) (0.3)
Collecting SQLAlchemy<3.0,>=1.4 (from kedro-datasets[pandas])
  Using cached SQLAlchemy-2.0.22-cp310-cp310-win_amd64.whl.metadata (9.7 kB)
Collecting deltalake>=0.10.0 (from kedro-datasets[pandas])
  Using cached deltalake-0.12.0-cp37-abi3-win_amd64.whl.metadata (5.1 kB)
Collecting lxml~=4.6 (from kedro-datasets[pandas])
  Using cached lxml-4.9.3-cp310-cp310-win_amd64.whl.metadata (3.9 kB)
Collecting openpyxl<4.0,>=3.0.6 (from kedro-datasets[pandas])
  Using cached openpyxl-3.1.2-py2.py3-none-any.whl (249 kB)
Collecting pandas-gbq<0.18.0,>=0.12.0 (from kedro-datasets[pandas])
  Using cached pandas_gbq-0.17.9-py2.py3-none-any.whl (25 kB)
Collecting pandas<3.0,>=1.3 (from kedro-datasets[pandas])
  Using cached pandas-2.1.1-cp310-cp310-win_amd64.whl.metadata (18 kB)
Collecting pyarrow>=6.0 (from kedro-datasets[pandas])
  Using cached pyarrow-13.0.0-cp310-cp310-win_amd64.whl.metadata (3.1 kB)
Collecting pyodbc~=4.0 (from kedro-datasets[pandas])
  Using cached pyodbc-4.0.39-cp310-cp310-win_amd64.whl (69 kB)
Collecting tables~=3.6.0 (from kedro-datasets[pandas])
  Using cached tables-3.6.1.tar.gz (4.6 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [21 lines of output]
      C:\Users\...\AppData\Local\Temp\pip-install-b4ce1dsx\tables_5b8b01ba41804417b71ba4bb574b9d6a\setup.py:18: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
        import pkg_resources
      * Using Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:34:57) [MSC v.1936 64 bit (AMD64)]
      * USE_PKGCONFIG: False
      * Found conda env: ``C:\Users\Yolan\mambaforge\envs\bug-kedro-pandas``
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\...\AppData\Local\Temp\pip-install-b4ce1dsx\tables_5b8b01ba41804417b71ba4bb574b9d6a\setup.py", line 634, in <module>
          libdir = compiler.has_function(package.target_function,
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\_distutils\ccompiler.py", line 893, in has_function
          objects = self.compile([fname], include_dirs=include_dirs)
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\_distutils\_msvccompiler.py", line 343, in compile
          self.initialize()
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\_distutils\_msvccompiler.py", line 253, in initialize
          vc_env = _get_vc_env(plat_spec)
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\msvc.py", line 233, in msvc14_get_vc_env
          return _msvc14_get_vc_env(plat_spec)
        File "C:\Users\...\mambaforge\envs\bug-kedro-pandas\lib\site-packages\setuptools\msvc.py", line 190, in _msvc14_get_vc_env
          raise distutils.errors.DistutilsPlatformError("Unable to find vcvarsall.bat")
      distutils.errors.DistutilsPlatformError: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

```
-- Separate them if you have more than one.
```

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Kedro version used (`pip show kedro` or `kedro -V`): ``kedro==0.18.14``
* Kedro plugin and kedro plugin version used (`pip show kedro-airflow`): ``kedro-datasets==1.7.1``
* Python version used (`python -V`): Python 3.10.12
* Operating system and version: Windows 11

## Potential fix: 

The usual suspect is :     
https://github.com/kedro-org/kedro-plugins/blob/b44bf0e9db43d1564e50bbc85c7a61676c1ddbe4/kedro-datasets/setup.py#L37-L42

And I confirm this is actually what's breaking. Funnily install ``pip install tables`` installs "tables=3.9.1" correctly (but I don't have the local setup to test the dataset).

### PR DESCRIPTION
**Pull request by Galileo-Galilei**
## Description

Close #402

## Development notes

- Removed Windows specific conditions in ``pandas.HDFDataSet`` extra dependencies in ``setup.py``
- (It seems my local ruff is slighlty reformatting the setup.py file, but I don't see any "dev dependencies specifying which black or ruff version I should use. The Lint job is passing so I let it as is)
- Tested locally on Windows 11: 

```bash
conda create -n tmp1 python=3.8 -y
conda activate tmp1
pip install ./kedro-datasets[pandas.HDFSDataSet]
```

```bash
conda create -n tmp1 python=3.9 -y
conda activate tmp2
pip install ./kedro-datasets[pandas.HDFSDataSet]
```

Both do not throw errors on install, unlike the current version. 

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [x] Added tests to cover my changes -> Local build test, but no unit test.

### CODE REVIEW

**Review by noklam — APPROVED** [review]
CI is passing so this seems to be a good change. https://github.com/kedro-org/kedro/pull/1139 is the PR that fix it, it was causing trouble to CI 2 years ago but not anymore.

File: kedro-datasets/setup.py [hunk 1]
```diff
@@ -36,9 +36,8 @@ def _collect_requirements(requires):
     "pandas.GBQQueryDataSet": [PANDAS, "pandas-gbq>=0.12.0, <0.18.0"],
     "pandas.HDFDataSet": [
         PANDAS,
-        "tables~=3.6.0; platform_system == 'Windows'",
-        "tables~=3.6, <3.9; platform_system != 'Windows' and python_version<'3.9'",
-        "tables~=3.6; platform_system != 'Windows' and python_version>='3.9'",
+        "tables~=3.6, <3.9; python_version<'3.9'",
+        "tables~=3.6; python_version>='3.9'",
```
Comment (astrojuanlu) [inline kedro-datasets/setup.py:40]: kedro-datasets now supports >= 3.9

```suggestion
        "tables~=3.6",
```

**Review by astrojuanlu — COMMENTED** [review]
Would love to see this merged

File: kedro-datasets/setup.py [hunk 2]
```diff
@@ -65,15 +64,24 @@ def _collect_requirements(requires):
     "plotly.JSONDataset": ["plotly>=4.8.0, <6.0"],
 }
 polars_require = {
-    "polars.CSVDataset": [POLARS],
-    "polars.EagerPolarsDataset":
-    [
-        POLARS, "pyarrow>=4.0", "xlsx2csv>=0.8.0", "deltalake >= 0.6.2"
+    "polars.CSVDataSet": [POLARS],
```
Comment (merelcht) [inline kedro-datasets/setup.py:67]:
```suggestion
    "polars.CSVDataset": [POLARS],
```

File: kedro-datasets/RELEASE.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Upcoming Release
 ## Major features and improvements
 ## Bug fixes and other changes
+* Removed Windows specific conditions in `pandas.HDFDataSet` extra dependencies
```
Comment (merelcht) [inline kedro-datasets/RELEASE.md:4]:
```suggestion
* Removed Windows specific conditions in `pandas.HDFDataset` extra dependencies
```

File: kedro-datasets/setup.py [hunk 4]
```diff
@@ -48,8 +48,7 @@ def _collect_requirements(requires):
     ],
     "pandas.HDFDataset": [
         PANDAS,
-        "tables~=3.6.0; platform_system == 'Windows'",
-        "tables~=3.6; platform_system != 'Windows'",
+        "tables~=3.6,
```
Comment (merelcht) [inline kedro-datasets/setup.py:51]:
```suggestion
        "tables~=3.6",
```

**Review by merelcht — APPROVED** [review]

**Pull request merged**